File: Linear_search/MATLAB/linear_search.m
arr = [145, 89, 2, 532, 8901, 8];
target = 2;
found = false;
index = -1;
n = length(arr);

for i = 1:n
  if arr(i) == target
    found = true;
    index = i;
    break;
  end
end

if found
  fprintf('The element %d was found at position %d\n', target, index);
else
  fprintf('The element %d was not found\n', target);
end

File: Linear_search/Erlang/linear_search.erl
-module(prog).
-export([main/0]).

main() ->
    Arr = [1235, 111, 63433, 2, 112],
    Target = 2,

    case search(Arr, Target) of 
      {found, Index} ->
          io:format("The element ~p was found at position ~p~n", [Target, Index]);
      not_found ->
        io:format("The element ~p was not found~p", [Target])
    end.

 search(List, Target) ->
    search(List, Target, 1).

 search([], _,_) -> not_found;
 search([Target | _], Target, Index) -> {found, Index};
 search([_ | Tail], Target, Index) -> search(Tail, Target, Index + 1).   
